### ISSUE TITLE
Port all modules from the JVM

### ISSUE THREAD

**Issue by Silv3rcircl3** [akkadotnet/Alpakka#1]
If possible we should port all available modules from the JVM.

- [x] AMQP
- [ ] Cassandra
- [ ] File
- [ ] MQTT

**Comment by alexvaluyskiy** [comment]
Why should we port all available modules from JVM? .NET world is too different from Java world

**Comment by SeanFarrow** [comment]
I'm happy to port the sqs/sns module as I'll be needing this in a project. would a pr be accepted?

**Comment by Silv3rcircl3** [comment]
@seanfarrow Sure !

**Comment by SeanFarrow** [comment]
Hi Marc,
How do we deal with the java/scala dsls, should we port as-is?
I notice they use Mokito when mocking things, what mocking framework do you prefer?

**Comment by Silv3rcircl3** [comment]
@seanfarrow I would say stay as close as possible to the jvm version, but if some things don't make sense in .net or need to be handled differently feel free to make those changes.

Regarding mocking, take what you like, we don't use mocking inside the core repo so we don't have a preference.

**Comment by SeanFarrow** [comment]
Ok, that’s fine, the only issue is the fact that the jvm version of the sqs stream things uses ElasticMQ which is java only to mock the sqs transport, should we just use the real one? If yes, what account should we use?

**Comment by Silv3rcircl3** [comment]
Is it possible to have a local instance you can use? If not, just port the tests, try it with some evaluation account and push it without automated tests.

**Comment by SeanFarrow** [comment]
The closest to a local instance is using ElasticMQ, let me see if we can run that from java in .net, may require a wcf service.

**Comment by Silv3rcircl3** [comment]
If it's to complicated, no worries, those connectors don't have the same "guarantees" as the core repo (see also the readme) and therefore don't neccarrily need a proper CI infrastructure.

**Comment by SeanFarrow** [comment]
Yes, but the project I’m working on need most things to be tested, so it would be nice to have that!
It looks pretty simple anyway, it just requires java installed and java home to be set up.
I’ll push without porting the tests first, then port the tests at a later date.
We can always leave the pull request open and merge certain things.
Cheers
Sean.

**Comment by Silv3rcircl3** [comment]
Sounds good!

**Comment by SeanFarrow** [comment]
Hi,

I’ve got the following scala code I’m trying to convert:
    Await.result(future, 1.second) mustBe publishResult :: Nil
How does this convert to c#/akka streams?
Cheers
Sean.

**Comment by Silv3rcircl3** [comment]
Task.Wait(TimeSpan.FromSeconds(1)).Should().BeTrue();
publishResult.Add(null);
Task.Result.Should().BeEquivalentTo(publishResult); 

Maybe the gitter room would be a better place for those questions ;-)

**Comment by SeanFarrow** [comment]
Just a thought, should we add some CI around this repo?
Given that we have tests, it would be good if we were able ot run things like akka.net does on pr's.

**Comment by Arkatufus** [comment]
There's a plan to set CI for Akka.net contrib, which is where this project will be moved, eventually.
Right now we're a bit busy maintaining 1.2.3 and preparing for 1.3.0

**Comment by SeanFarrow** [comment]
Happy to help maintain this as we’re using/converting a lot of the scala code,  if an extra pair of hands would be useful.

**Comment by SeanFarrow** [comment]
I've just been looking at this again and noticed the .net version of the sns code uses a string and the jvm version uses a PublishRequest. Should we change this?

By not using a PublishRequest, you lose the ability to provide metadata.
